### PR TITLE
Add parameter to control run-wise correlations

### LINKED ISSUE #1543: Add option to not write out run-wise correlation matrices
## Summary

Don't write out run-wise correlation matrices in abcd or hbcd mode.
Do write them out in linc mode.
Raise an error in none mode if it's not provided.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #1543.

## Changes proposed in this pull request

- Create new parameter, `--output-run-wise-correlations [auto|y|n]`, which determines if run-level correlation matrices are written out or not. For abcd and hbcd mode, this is disabled by default. For linc and nichart mode, it is enabled.

---

> [!NOTE]
> Add --output-run-wise-correlations [auto|y|n] with mode-specific defaults and conditionally compute/write run-wise correlation matrices based on this flag and whether multiple runs exist.
> 
> - **CLI**:
>   - Add `--output-run-wise-correlations/--output_run_wise_correlations` (auto|y|n) to control writing run-level correlation matrices.
>   - Per-mode defaults in `_validate_parameters`: `abcd/hbcd`→`False`, `linc/nichart`→`True`, `none` requires explicit value; add assertion.
> - **Config**:
>   - Add `workflow.output_run_wise_correlations` and include in `DEFAULT_CONFIG_HASH_FIELDS`.
> - **Workflows**:
>   - Propagate `has_multiple_runs` through `init_postprocess_{nifti,cifti}_wf`, connectivity, and outputs.
>   - Gate correlation computation and sinks in `connectivity.py` and `outputs.py` when `'all' in correlation_lengths` AND (`output_run_wise_correlations` OR not `has_multiple_runs`).
> - **Tests/Fixtures**:
>   - Update expected outputs to reflect conditional omission of run-wise correlations.
>   - Add CLI arg in tests; enforce requirement in `none` mode; pass `has_multiple_runs=False` in connectivity tests.
>   - Update hashed filenames where config hash changes.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9b2fab1b441c6d4e1c89aaf5f8fc61e91d156ee4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by mattcieslak — APPROVED** [review]

File: xcp_d/tests/data/test_pnc_cifti_t2wonly_outputs.txt [hunk 1]
```diff
@@ -15,67 +15,67 @@ logs/CITATION.html
...
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-coverage_bold.json
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-coverage_bold.tsv
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-mean_timeseries.json
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-mean_timeseries.tsv
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-pearsoncorrelation_relmat.json
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-pearsoncorrelation_relmat.tsv
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-reho_bold.json
+sub-1648798153/ses-PNC1/func/sub-1648798153_ses-PNC1_hash-e083b4a2_task-idemo_space-fsLR_seg-MIDB_stat-reho_bold.tsv
+sub-1648798153_hash-e083b4a2.html
+sub-1648798153_ses-PNC1_hash-e083b4a2_executive_summary.html
```
[the remark this answers is not in the document]
Comment (tsalo) [reply xcp_d/tests/data/test_pnc_cifti_t2wonly_outputs.txt]: The new field, `output_run_wise_correlations`, is now one of the config fields that affects the hash.
